### DISCUSSION TITLE
issues with display

### DISCUSSION THREAD

**Discussion by ramonflores08** [OMOTE-Community/OMOTE-Firmware#18] [General]
I uploaded the code and it seems that the screen is not displaying properly. Im not sure if its a hardware issue or software.  Below is an image of what i see. The screen seems to be working and the touch works fine. If i press buttons on the screen the image of the button i pressed gets fixed. Let me know what you think.
![20230726_002352](https://github.com/CoretechR/OMOTE/assets/7199615/8adc0053-e4bc-4dca-a711-ee5933e2aaca)

**Comment by CoretechR** [comment]
Hi, great that (almost) everything seems to work. Could you try to lower the SPI frequency from 80 to 40 MHz? Its barely noticeable in terms of frame rate but some displays might not be able to handle the full speed.

**Reply by ramonflores08** [reply]
Works as expected now!

**Comment by ramonflores08** [comment]
Assembled and ready to test! I was beginning to doubt my soldering skills when I saw the screen.

![20230726_014923](https://github.com/CoretechR/OMOTE/assets/7199615/07585e00-fd1c-415b-b717-71de27e9b6a8)

**Reply by CoretechR** [reply]
Awesome! I should probably set the default speed to 40MHz. 
Please let me know if there are any other issues. This is only the third, fully functional board 😄

**Reply by ramonflores08** [reply]
I believe i may have done something wrong with my soldering. I shall take a look again since it shows the usb symbol on the top  right even when its disconnected. I will double check all the connections tomorrow.

**Reply by phant0mias** [reply]
Hello ramonflores08,

the PCB is coming soon, the only thing that is missing is the display.

Which display have you used? Is there a part number on the display like DT280QV10-CT or CH280QV10-CT. 

Can you tell me where you have ordered it?

Thank you for your answer.

**Reply by Dark1886** [reply]
@phant0mias I used [This Screen](https://www.adafruit.com/product/2770?gclid=CjwKCAjwt52mBhB5EiwA05YKo2lW5xmLPy0jpRNeeNK5eGd3-7hEH26Cy0PIzrrvfucoCJlAK-3mHBoCyJ0QAvD_BwE) and you can order it from multiple sites, it would depend what supplier is most convenient for your location.

**Reply by phant0mias** [reply]
Thanks for the answer, can you tell me the part number of the display. It seems there are different sold by adafruit.
DT280QV10-CT or CH280QV10-CT?

**Reply by Dark1886** [reply]
@phant0mias I guess I'm not sure what one I got. I just ordered the PN:2770 from adafruit. From the datasheet on their website, the numbering system defines the "CH" as "Shenzhen ChengHao Optoelectronic Co., Ltd", So it's possible they just changed suppliers at some point and changed the part number for easier tracking.

**Reply by phant0mias** [reply]
Yes they changed the supplier in january. I heared about multitouch problems with the CH280QV10-CT display, but I am not sure. I´m interested which display are already used and running without problems

**Reply by Dark1886** [reply]
@phant0mias I'll have to defer to others with their experiences. I haven't had mine long enough to be comfortable saying it is working "without problems".

**Reply by phant0mias** [reply]
and the part number (is on the back side)?

**Reply by Dark1886** [reply]
@phant0mias Honestly not sure. I already stuck it to my PCB and don't want to peel it back off.

**Reply by ramonflores08** [reply]
Same, I don't want to peel it off. I ordered mine off of adafruit though.

https://www.adafruit.com/product/2770

**Reply by CoretechR** [reply]
The old displays and boards from Adafruit (until ca. January 2023) are type DT280QV10-CT. One of these was used for my working prototype. The new displays and boards should have the label CH280QV10, meaning they are made by ChengHao Optoelectronic. I ordered three of these new displays directly from the manufacturer earlier this year. They all had problems with the touch sensitivity (random ghost touches). I contacted them but was not able to resolve the issue yet. 
So for the moment, if you want guaranteed working displays, I can only recommend buying older Adafruit boards (blue solder mask). The newer ones might work, too, but I have not tested one directly from Adafruit yet.

**Reply by CoretechR** [reply]
@ramonflores08 It would be very interesting to find out what type your display is. The silkscreen on the flex cable is slightly different. The left one is an older Adafruit board (most likely DT280QV10) and the right display is a Chenghao type. 

![IMG_8199](https://github.com/CoretechR/OMOTE/assets/13223470/458814b8-6abc-447d-a108-1ae1662d4e92)

**Reply by ramonflores08** [reply]
I shall take one for the team and disassemble it lol. Give me a few mins.

**Comment by ramonflores08** [comment]
This is the back of my screen. It is the newer one you mentioned from ChengHao Optoelectronic. It has worked with no issues ever since I changed the SPI frequency from 80 to 40 MHz as per the previous comments.

![20230731_120809](https://github.com/CoretechR/OMOTE/assets/7199615/87f1ebe2-3515-408a-a3de-6134e3e3f7d1)

**Comment by phant0mias** [comment]
so it is the Chenghao type.

It works with no problems?

**Reply by ramonflores08** [reply]
Well, I did have to lower the frequency, but yeah no problems. It feels snappy, and I have not noticed any issues.

**Comment by CoretechR** [comment]
@ramonflores08 Thank you for the sacrifice! 😄 
That gives me hope, that there is no general problem with the Chenghao displays. Maybe it was just a bad batch that I received.

**Reply by phant0mias** [reply]
Thnaks for opening the remote control. This are good news

**Reply by ramonflores08** [reply]
Could be. Sorry about the bad luck. I honestly thought I had a bad screen as well when I saw the image. I have had not noticed any ghost inputs. I think I have narrowed down my charger status to component U1. It keeps outputting approximately 1V even when it's disconnected from the USB. I am going to get a new one and replace it.

**Reply by CoretechR** [reply]
The STAT pin should be high while battery powered and low while charging. For the USB icon to appear, the STAT pin must be either low (charging) or it must be high while the supply voltage is above 4.3V (charge completed but plugged in). Are you using the original MCP73831?
![grafik](https://github.com/CoretechR/OMOTE/assets/13223470/fe2bc5f5-7ed7-4bde-8e09-14013b9434c7)

**Reply by ramonflores08** [reply]
I belive i got the correct one. below is the link to the one I ordered.

https://www.mouser.com/ProductDetail/579-MCP73831T-2ATIOT

**Reply by CoretechR** [reply]
Yep, that's the correct part.

**Reply by ramonflores08** [reply]
I am going to replace the U1 IC since I do hear a high pitch wine when plugging in the USB. By removing it the board acts normally with the VBUS going back up to the expected 4.3V. I may have just gotten the luck of the draw and received a faulty chip, or I damaged it who knows?

**Comment by samdowrick** [comment]
I also have the new adafruit display and my touch functionality is not working.

**Reply by phant0mias** [reply]
We have several new displays and everything is working fine. Have you checked the error messages in the terminal windows of platformio? 
I am not absolutly sure, but touch is using I2C. Maybe there are error messages

**Reply by samdowrick** [reply]
I'm getting the timeout error. The accelerometer isn't working either

**Reply by phant0mias** [reply]
so the problem is I2C an not the display

**Reply by samdowrick** [reply]
It would seem to be. I've checked all solder joints and traces and everything is okay. It seems to be a software issue

**Reply by samdowrick** [reply]
It would seem to be. I've checked all solder joints and traces and everything is okay. It seems to be a software issue

**Reply by CoretechR** [reply]
@samdowrick You could try running a i2c scan that lists all the i2c addresses. This would rule out a software problem.

**Reply by samdowrick** [reply]
Funny you say that, I'm actually just about to. I'll let you know how it goes

**Reply by samdowrick** [reply]
> @samdowrick You could try running a i2c scan that lists all the i2c addresses. This would rule out a software problem. 

So I ran the scan and no devices found. I'm like 99% sure that everything is soldered correctly

**Reply by samdowrick** [reply]
Just resoldered the esp32 and the accelerometer and I'm now finding the devices

**Reply by samdowrick** [reply]
Just resoldered the esp32 and the accelerometer and I'm now finding the devices

**Reply by samdowrick** [reply]
Fully working, must have been a short on one for the chips
